datalad/local/wtf.py

```diff
@@ -138,9 +139,33 @@ def _describe_system():
                                   _t2s(pl.win32_ver())]).rstrip(),
         'max_path_length': get_max_path_length(getpwd()),
         'encoding': get_encoding_info(),
+        'filesystem': {l: _get_fs_type(l, p) for l, p in
+                       [('CWD', os.getcwd()),
+                        ('TMP', tempfile.gettempdir()),
+                        ('HOME', str(Path.home()))]}
     }
 
 
+def _get_fs_type(loc, path):
+    global fs_hint
+    fs_hint = None
+    try:
+        from psutil import disk_partitions
+        match = ""
+        fs = ""
+        for part in disk_partitions():
+            if path.startswith(part.mountpoint) and \
+                    len(match) < len(part.mountpoint):
+                fs = part.fstype
+                match = part.mountpoint
```

Comment:
looking at this trove of information:
```
In [10]: disk_partitions()[0]
Out[10]: sdiskpart(device='/dev/nvme0n1p5', mountpoint='/', fstype='ext4', opts='rw,relatime,errors=remount-ro', maxfile=255, maxpath=4096)
```
I want to say -- **I want it all** (besides `device` may be -- don't think it is relevant).
- `maxfile`/`maxpath` -- we even try to deduce max filename length, could be critical to know "actual" ones known for file system
- `opts` - most of the times I report anything filesystems related to @joeyh - we end up wanting to see the mount options